controllers/suite_test.go

```diff
@@ -205,6 +206,11 @@ func TestMain(m *testing.M) {
 		panic(fmt.Sprintf("Failed to create a test registry server: %v", err))
 	}
 
+	fg := feathelper.FeatureGates{}
+	fg.SupportedFeatures(map[string]bool{
+		features.GitManagedTransport: true,
+	})
```

Comment:
I think this would be better if aligned with what we do at `main.go` and simply rely on the existing default values, so as they change our tests would automatically capture/validate such changes.

```golang
        fg := feathelper.FeatureGates{}
	fg.SupportedFeatures(features.FeatureGates())
```

https://github.com/fluxcd/source-controller/blob/main/main.go#L153-L154

Reply:
Revisiting this, looking at the code again, I see that this does affect the default reconcilers that we run in the test environment. So, it affects how `TestGitRepositoryReconciler_Reconcile()` behaves.
I'd support @pjbgf 's recommendation to do the same that's done in main.go to ensure we don't have different configurations in tests and main code.